### DISCUSSION TITLE
Students charged for github codespaces when using classroom

### DISCUSSION THREAD

**Discussion by powerjg** [community/community#88769] [GitHub Education] [answered]
### Hi everyone,

According to [the documentation](https://docs.github.com/en/education/manage-coursework-with-github-classroom/integrate-github-classroom-with-an-ide/using-github-codespaces-with-github-classroom#about-github-codespaces) GitHub Codespaces when used with GitHub Classroom should charge the organization for usage, not the student. However, that is not happening right now. I have students in my class that have "used up" their 120 free hours.

> Note: If students use GitHub Codespaces in their assignment repositories for GitHub Classroom, the usage will be charged to the organization that owns the classroom, and will not affect the usage for a student's personal account.

Have I misconfigured something? I can give links to the classroom, etc. if that helps.

Thanks for the support!

**Comment by mishmanners** [comment]
Thanks for reaching out @powerjg. When using Codespaces, the students need to ensure that when they are accessing the Codespace from the organisation, and that the Codespace is showing that the organisation is paying for it. This is a good way to teach students about correctly navigating to the repo and not forking it and using it on their account.

This screenshot shows what the students should be seeing:
![Codespaces usage](https://github.com/community/community/assets/36594527/f3f8fa25-2c45-4f6e-803d-6d7e4b7c0202)

When teaching the lesson, make sure the students can see on their screens that your org is paying for the Codespace. If you want to enable students to do their own things with Codespaces, you'll need to add all the students to the org you are using instead of having them fork the repo to their own account. Since it's GitHub Classroom, make sure the classroom is set up on the organisation that you've setup Codespaces for, and ensure students are working from the Classroom.

Are there any specific errors or ways that students are accessing the Codespace through their personal accounts?

**Comment by powerjg** [comment]
Does this mean the students need to be "invited" to the organization and cannot be "outside collaborators"?

Is there an easy way in github classroom to automatically do this?

I just tested with a test personal account, and I see that it *does not say* "paid for by `<my organization>`". It says "paid for by `<my test account>`."

Edit: I just tested with my test account. I invited myself to the repo, but it still says "Codespace usage for this repository is paid for by `<my test account>`". I'm not sure what I'm doing wrong :)

**Comment by powerjg** [comment, marked as answer]
Aha! I believe I figured out the problem. For the organization that I created for the classroom, the default was for codespace ownership to be "user" not "organization". Changing this radio button seemed to have worked.

<img width="729" alt="image" src="https://github.com/community/community/assets/109596/5daaf663-40aa-47c7-be71-7a74c612e140">

**Reply by te3WAxGF** [reply]
> Aha! I believe I figured out the problem. For the organization that I created for the classroom, the default was for codespace ownership to be "user" not "organization". Changing this radio button seemed to have worked.
> 
> <img alt="image" width="729" src="https://private-user-images.githubusercontent.com/109596/298205951-5daaf663-40aa-47c7-be71-7a74c612e140.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODIyNjY1NjIsIm5iZiI6MTc4MjI2NjI2MiwicGF0aCI6Ii8xMDk1OTYvMjk4MjA1OTUxLTVkYWFmNjYzLTQwYWEtNDdjNy1iZTcxLTdhNzRjNjEyZTE0MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDYyNFQwMTU3NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZTJhNmQ4ZTRhNTc5M2EwNGRmZGJjYmQ3Mjc0NzNlZDE3ZmM3OWIwMzk3NWUwMTkwYjNmODdjYTU4Y2UwYTdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.3FLmN8J6VfDeAkQJb2n8J19TEH22-h8pLX6NbX2funA">
